### PR TITLE
Layout editor list dialog error fix

### LINKED ISSUE #6984: error message on changing table in demographic forms
### Describe the bug
admin/forms/layouts - demographics. Change a list. An error message is displayed

concise description of what the bug is:
if one clicks on the list field, to change the list to be used, an error message is displayed in a pop up window in the 'select list' window. It varies from system to system, examples are 

 `DataTables warning: table id=my_data_table - Requested unknown parameter '2' for row 0, column 2. For more information about this error, please see http://datatables.net/tn/4`
and
`undefined
URL: undefined
Line: undefined Column: undefined
Error object: undefined`

If one clicks 'ok' to each error message then one is finally able to change the table

### To Reproduce

image

1. choose menu admin/forms/layouts 
2.  use drop down to choose 'demographics'. 
3. click on a 'list' field to change the list
5. click 'ok' until errors stop displaying
6. procede to change the list as desired

### Expected behavior

one should be able to change the list without error messages displaying first

### Client configuration

- Browser: firefox
- OpenEMR version: 7.0.1(1) and  7.0.2-dev
- Operating system: linux 

possibly other OS and browser, i can find out if necessary

### PR DESCRIPTION
**Pull request by sjpadgett**
- add modifier to lists search returned array

Fixes #6984 

#### Short description of what this resolves:
- see issue
Apparently datatables project is now enforcing all columns from table build has a data object.
In the past we default to the code type header for lists (three columns) however failed to populate the modifier column on search return results.

### CODE REVIEW

**Comment by sjpadgett** [comment]
This one had me scratching my head for a minute. The reporting error is somewhat cryptic!

File: interface/login/login.php [hunk 1]
```diff
@@ -83,11 +83,13 @@
...
         $div_app = sprintf('<input type="hidden" name="appChoice" value="%s">', attr($_REQUEST['app']));
     } else {
+        $opt_htm = '';
+        // Is $opt_default suppose to be defined somewhere?
         foreach ($emr_app as $opt_disp => $opt_value) {
             $opt_htm .= sprintf(
                 '<option value="%s" %s>%s</option>\n',
                 attr($opt_disp),
-                ($opt_disp == $opt_default ? 'selected="selected"' : ''),
+                ($opt_disp == ($opt_default ?? '') ? 'selected="selected"' : ''),
```
Comment (bradymiller) [inline interface/login/login.php:92]: `$opt_default` should be `$emr_app_def`
(per above code)

File: interface/login/login.php [hunk 2]
```diff
@@ -83,11 +83,13 @@
...
                 text(xl_list_label($opt_disp))
```
Comment (bradymiller) [inline interface/login/login.php:93]: also above line 91 should use `$opt_value` and not `$opt_disp`
(per code above)

Comment (bradymiller) [reply interface/login/login.php:93]: clarify (for above line):
```
text(xl_list_label($opt_value))
```

**Comment by sjpadgett** [comment]
Thanks @bradymiller you made take a closer look at this! lol well maybe thanks as this feature wasn't included to be rendered in the login twig template!
For those curious this is a feature that @mdsupport put in way back that allows a menu to support redirecting login to a specific page as an different application login.

**Comment by sjpadgett** [comment]
I cleaned this up to now work if somebody wants to invest the time to add to twig. This hasn't worked since v6.1.0 with nobody complaining it is missing! I think since we have modules now this feature is better served using an event so IMO this feature can go.

**Comment by bradymiller** [comment]
I just brought this into the rel-702 branch
(plan to bring in commits from master to rel-702 quickly (and in order of master) prior to release to avoid missing anything and avoiding merging issues)

**Comment by mdsupport** [comment]
> Thanks @bradymiller you made take a closer look at this! lol well maybe thanks as this feature wasn't included to be rendered in the login twig template! For those curious this is a feature that @mdsupport put in way back that allows a menu to support redirecting login to a specific page as an different application login.



> I cleaned this up to now work if somebody wants to invest the time to add to twig. This hasn't worked since v6.1.0 with nobody complaining it is missing! I think since we have modules now this feature is better served using an event so IMO this feature can go.

@sjpadgett, in case you want to complete the cleanup you can leverage PR #4821 we submitted sometime ago. Apart from the module approach, our [mdpub repo](https://github.com/mdsupport/mdpub) provides less intrusive approach.

**Pull request merged**